src/overlays/actors/ovl_En_Ma_Yto/z_en_ma_yto.c

```diff
@@ -21,126 +83,1425 @@ const ActorInit En_Ma_Yto_InitVars = {
+            break;
+
+        default:
+            EnMaYto_SetupDefaultWait(this);
+            break;
+    }
+}
+
+s32 EnMaYto_SearchRomani(EnMaYto* this, GlobalContext* globalCtx) {
+    Actor* npcActor = globalCtx->actorCtx.actorList[ACTORCAT_NPC].first;
+
+    while (npcActor != NULL) {
+        if (npcActor->id == ACTOR_EN_MA_YTS) {
+            EnMaYts* romani = (EnMaYts*)npcActor;
+            s16 romaniType;
+
+            romaniType = EN_MA_YTS_PARSE_TYPE(&romani->actor);
+            if ((this->type == MA_YTO_TYPE_DINNER && romaniType == MA_YTS_TYPE_SITTING) ||
+                (this->type == MA_YTO_TYPE_BARN && romaniType == MA_YTS_TYPE_BARN)) {
+                this->actor.child = &romani->actor;
+                romani->actor.parent = &this->actor;
+                return true;
+            } else {
+                npcActor = npcActor->next;
+                continue;
+            }
+        }
+        npcActor = npcActor->next;
+    }
+
+    return false;
+}
+
+/**
+ * Returns:
+ * - 2: Romani was found.
+ * - 1: Romani was not found.
+ * - 0: There's no need to find Romani.
+ */
+s32 EnMaYto_TryFindRomani(EnMaYto* this, GlobalContext* globalCtx) {
+    switch (this->type) {
+        case MA_YTO_TYPE_DEFAULT:
+            return 0;
+
+        case MA_YTO_TYPE_DINNER:
+            if (!(gSaveContext.weekEventReg[0x16] & 1) && CURRENT_DAY == 2) {
+                return 0;
+            }
+            if (EnMaYto_SearchRomani(this, globalCtx)) {
+                return 2;
+            }
+            return 1;
+
+        case MA_YTO_TYPE_BARN:
+            // if (AliensDefeated)
+            if (gSaveContext.weekEventReg[0x16] & 1) {
+                if (EnMaYto_SearchRomani(this, globalCtx)) {
+                    return 2;
+                }
+                return 1;
+            }
+            return 0;
+
+        case MA_YTO_TYPE_AFTERMILKRUN:
+            return 0;
+
+        case MA_YTO_TYPE_4:
+            return 0;
+    }
+
+    return 0;
+}
+
+void EnMaYto_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnMaYto* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnMaYto_SetupKeepLookingForRomani(EnMaYto* this) {
+    this->actionFunc = EnMaYto_KeepLookingForRomani;
+}
+
+void EnMaYto_KeepLookingForRomani(EnMaYto* this, GlobalContext* globalCtx) {
+    if (EnMaYto_TryFindRomani(this, globalCtx) == 2) {
+        EnMaYto_ChooseAction(this, globalCtx);
+    }
+}
+
+void EnMaYto_SetupDefaultWait(EnMaYto* this) {
+    if (this->actor.shape.rot.y == this->actor.home.rot.y) {
+        this->currentAnim = 11;
+        EnMaYto_ChangeAnim(this, 11);
+    } else {
+        this->currentAnim = 1;
+        EnMaYto_ChangeAnim(this, 1);
+    }
+
+    EnMaYto_InitFaceExpression(this);
+    this->unk_31E = 2;
+    this->actionFunc = EnMaYto_DefaultWait;
+}
+
+void EnMaYto_DefaultWait(EnMaYto* this, GlobalContext* globalCtx) {
+    s16 rotY = this->actor.home.rot.y - 0x8000;
```

Comment:
Just a highlight, but it's interesting that they use `- 0x8000` instead of `+ 0x8000` to turn around 180 degrees here. Makes the `BINANG_ROT180` macro a little less powerful

Reply:
Yeah, it is a shame, but the whole codebase is kinda inconsistent :c